### PR TITLE
Improve pd recover

### PR DESCRIPTION
**Pull request by yiduoyunQ**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-tidb-operator) that's required for repo owners to accept my contribution.

### What is changed, added, or deleted? (Required)
Current [pd-recover](https://docs.pingcap.com/tidb-in-kubernetes/stable/pd-recover) doc steps create new clear PD pod to establish a new PD cluster, this method will lose all configuration that has taken effect in previous PD cluster.  If customer use [Placement Rules](https://docs.pingcap.com/tidb/stable/placement-rules-in-sql) feature and lose rules after pd recover, it will take great impact to the TiDB cluster.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.4 (TiDB Operator 1.4 versions)
- [x] v1.3 (TiDB Operator 1.3 versions)
- [x] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): 
https://github.com/pingcap/docs-cn/blob/b9685f4c59b42cab29d02e3566182aa89b6ae6e7/pd-recover.md

### CODE REVIEW

File: zh/pd-recover.md [hunk 1]
```diff
@@ -31,9 +31,211 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
 
     `pd-recover` 在当前目录下。
 
-## 使用 PD Recover 恢复 PD 集群
+## 使用 PD Recover 恢复 PD 集群（有可用 PD 节点）
```
Comment (ran-huang) [inline zh/pd-recover.md:34]:
```suggestion
## 场景 1：集群中有可用 PD 节点
```

File: zh/pd-recover.md [hunk 2]
```diff
@@ -31,9 +31,211 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
 
-本小节详细介绍如何使用 PD Recover 来恢复 PD 集群。
+本小节详细介绍如何使用 PD Recover 并通过可用 PD 节点来恢复 PD 集群。
```
Comment (ran-huang) [inline zh/pd-recover.md:36]:
```suggestion
本小节详细介绍如何使用 PD Recover 并通过可用的 PD 节点来恢复 PD 集群。本小节内容仅适用于集群中有可用的 PD 节点，如集群中所有 PD 节点均故障且无法恢复，请参考[这篇文档](#场景-2-所有-PD-节点都故障且无法恢复)。
```

File: zh/pd-recover.md [hunk 3]
```diff
@@ -31,9 +31,211 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+使用以下命令重启 TiDB 和 TiKV 实例：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl delete pod -l app.kubernetes.io/component=tidb,app.kubernetes.io/instance=${cluster_name} -n ${namespace} &&
+kubectl delete pod -l app.kubernetes.io/component=tikv,app.kubernetes.io/instance=${cluster_name} -n ${namespace}
+```
+
+## 使用 PD Recover 恢复 PD 集群（所有 PD 都故障且无法恢复）
```
Comment (ran-huang) [inline zh/pd-recover.md:232]:
```suggestion
## 场景 2：所有 PD 节点都故障且无法恢复
```

File: zh/pd-recover.md [hunk 4]
```diff
@@ -31,9 +31,211 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+本小节详细介绍如何使用 PD Recover 并通过新建 PD 的方式来恢复 PD 集群。
```
Comment (ran-huang) [inline zh/pd-recover.md:234]:
```suggestion
本小节详细介绍如何使用 PD Recover 并通过新建 PD 的方式来恢复 PD 集群。本小节内容仅适用于集群中所有 PD 节点均故障且无法恢复的情况，如集群中有可用的 PD 节点，请参考[这篇文档](#场景-1-集群中有可用-PD-节点)。
```

File: zh/pd-recover.md [hunk 5]
```diff
@@ -31,9 +31,211 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+>
+> 通过可用 PD 节点来恢复集群，可以保留之前 PD 已生效的所有配置信息。
+
+### 第 1 步：恢复 PD 集群 Pod
+
+1. 使用一个可用 PD 节点强制重建 PD 集群。
+   
+    > **提示：**
+    >
+    > 这里以 pd-0 为例，若使用其他 PD pod 请调整对应的命令。
```
Comment (ran-huang) [inline zh/pd-recover.md:48]:
```suggestion
> **提示：**
>
> 这里以 pd-0 为例，若使用其他 PD pod 请调整对应的命令。
    
使用一个可用 PD 节点 `pd-0` 强制重建 PD 集群。具体步骤如下：
```

File: zh/pd-recover.md [hunk 6]
```diff
@@ -31,9 +31,211 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+    让 pd-0 pod 进入 Debug 模式：
```
Comment (ran-huang) [inline zh/pd-recover.md:50]:
```suggestion
1. 让 pd-0 pod 进入 Debug 模式：
```

File: zh/pd-recover.md [hunk 7]
```diff
@@ -31,9 +31,211 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl annotate pod ${cluster_name}-pd-0 -n ${namespace} runmode=debug
+    kubectl exec ${cluster_name}-pd-0 -n ${namespace} -- kill -SIGTERM 1
+    ```
+
+    进入 pd-0 pod：
```
Comment (ran-huang) [inline zh/pd-recover.md:59]:
```suggestion
2. 进入 pd-0 pod：
```

File: zh/pd-recover.md [hunk 8]
```diff
@@ -31,9 +31,211 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl -n ${cluster_name} exec -it basic-pd-0 -- sh
+    ```
+
+    参考默认启动脚本 [_start_pd.sh.tpl](https://github.com/pingcap/tidb-operator/blob/master/charts/tidb-cluster/templates/scripts/_start_pd.sh.tpl)，在 pd-0 里配置环境变量：
```
Comment (ran-huang) [inline zh/pd-recover.md:67]:
```suggestion
3. 参考默认启动脚本 [`_start_pd.sh.tpl`](https://github.com/pingcap/tidb-operator/blob/master/charts/tidb-cluster/templates/scripts/_start_pd.sh.tpl)，在 pd-0 里配置环境变量：
```

File: zh/pd-recover.md [hunk 9]
```diff
@@ -31,9 +31,211 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+    then
+    until result=$(wget -qO- -T 3 http://${discovery_url}/new/${encoded_domain_url} 2>/dev/null); do
+    echo "waiting for discovery service to return start args ..."
+    sleep $((RANDOM % 5))
+    done
+    ARGS="${ARGS}${result}"
+    fi
+    ```
+
+    然后使用原始的 pd-0 数据目录强制启动一个新的 PD 集群：
```
Comment (ran-huang) [inline zh/pd-recover.md:129]:
```suggestion
4. 使用原始的 pd-0 数据目录强制启动一个新的 PD 集群：
```

File: zh/pd-recover.md [hunk 10]
```diff
@@ -31,9 +31,211 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+    sleep $((RANDOM % 10))
+    echo "/pd-server --force-new-cluster ${ARGS}"
+    exec /pd-server --force-new-cluster ${ARGS} &
+    ```
+
+    退出 pd-0 pod:
+
+    ```shell
+    exit
+    ```
```
Comment (ran-huang) [inline zh/pd-recover.md:144]:
```suggestion
5. 退出 pd-0 pod：

    ```shell
    exit
    ```
```

File: zh/pd-recover.md (hunk 9 above)
Comment (ran-huang) [inline zh/pd-recover.md:129]: 这里用户要如何确认新的 PD 集群成功启动？能否说明一下

Comment (yiduoyunQ) [reply zh/pd-recover.md]: 参考原来的确认步骤，确认 pd-0 日志里的 welcome

Comment (yiduoyunQ) [reply zh/pd-recover.md]: done

**Review by ran-huang — APPROVED** [review]

**Comment by ran-huang** [comment]
@csuzhangxc Could you please review the Chinese part?

File: zh/pd-recover.md [hunk 11]
```diff
@@ -31,9 +31,193 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
 
     `pd-recover` 在当前目录下。
 
-## 使用 PD Recover 恢复 PD 集群
+## 场景 1：集群中有可用 PD 节点
 
-本小节详细介绍如何使用 PD Recover 来恢复 PD 集群。
+本小节详细介绍如何使用 PD Recover 并通过可用的 PD 节点来恢复 PD 集群。本小节内容仅适用于集群中有可用的 PD 节点，如集群中所有 PD 节点均故障且无法恢复，请参考[这篇文档](#场景 2：所有 PD 节点都故障且无法恢复)。
```
Comment (csuzhangxc) [inline zh/pd-recover.md:36]: the link here is not working.

File: zh/pd-recover.md [hunk 12]
```diff
@@ -31,9 +31,193 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+    ```shell
+    kubectl annotate pod ${cluster_name}-pd-0 -n ${namespace} runmode=debug
+    kubectl exec ${cluster_name}-pd-0 -n ${namespace} -- kill -SIGTERM 1
+    ```
+
+2. 进入 pd-0 pod：
+
+    ```shell
+    kubectl -n ${cluster_name} exec -it basic-pd-0 -- sh
```
Comment (csuzhangxc) [inline zh/pd-recover.md:60]:
```suggestion
    kubectl exec ${cluster_name}-pd-0 -n ${namespace} -it -- sh
```

**Review by csuzhangxc — COMMENTED** [review]
Great work!

File: zh/pd-recover.md (hunk 11 above)
Comment (yiduoyunQ) [reply zh/pd-recover.md:36]: 我在 Markdown 编辑器里预览时，可以跳转，# 跳转到下面场景2 的标题

File: zh/pd-recover.md [hunk 13]
```diff
@@ -31,9 +31,193 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+    ```
+
+3. 参考默认启动脚本 [`_start_pd.sh.tpl`](https://github.com/pingcap/tidb-operator/blob/master/charts/tidb-cluster/templates/scripts/_start_pd.sh.tpl)，在 pd-0 里配置环境变量：
```
[the remark this answers is not in the document]
Comment (yiduoyunQ) [reply zh/pd-recover.md]: done

File: zh/pd-recover.md (hunk 11 above)
Comment (csuzhangxc) [reply zh/pd-recover.md:36]: https://github.com/yiduoyunQ/docs-tidb-operator/blob/improve-pd-recover/zh/pd-recover.md#%E5%9C%BA%E6%99%AF-1%E9%9B%86%E7%BE%A4%E4%B8%AD%E6%9C%89%E5%8F%AF%E7%94%A8-pd-%E8%8A%82%E7%82%B9

Comment (yiduoyunQ) [reply zh/pd-recover.md:36]: @ran-huang 帮忙确认下，这个应该没问题哇，不清楚 github 里要怎么展示跳转

Comment (ran-huang) [inline zh/pd-recover.md:36]:
```suggestion
本小节详细介绍如何使用 PD Recover 并通过可用的 PD 节点来恢复 PD 集群。本小节内容仅适用于集群中有可用的 PD 节点，如集群中所有 PD 节点均故障且无法恢复，请参考[这篇文档](#场景-2所有-pd-节点都故障且无法恢复)。
```

Comment (ran-huang) [reply zh/pd-recover.md]: fixed~

**Review by csuzhangxc — APPROVED** [review]

File: en/pd-recover.md [hunk 14]
```diff
@@ -31,9 +31,193 @@ PD Recover is a disaster recovery tool of [PD](https://docs.pingcap.com/tidb/sta
 
     `pd-recover` is in the current directory.
 
-## Recover the PD cluster
+## Scenario 1: At least one PD node is alive
 
-This section introduces how to recover the PD cluster using PD Recover.
+This section introduces how to recover the PD cluster using PD Recover and alive PD nodes. This section is only applicable to the scenario where the PD cluster has alive PD nodes. If all PD nodes are unavailable, refer to [Scenario 2](#scenario-2-all-pd-nodes-are-unavailable).
```
Comment (ran-huang) [inline en/pd-recover.md:36]:
```suggestion
This section introduces how to recover the PD cluster using PD Recover and alive PD nodes. This section is only applicable to the scenario where the PD cluster has alive PD nodes. If all PD nodes are unavailable, refer to [Scenario 2](#scenarios-2-all-pd-nodes-are-down-and-cannot-be-recovered).
```

**Pull request merged**